src/components/Header/Header.tsx

```diff
@@ -1,18 +1,23 @@
 import Logo from "@assets/logo.svg";
 import Cart from "@assets/cart.svg";
-import { HeaderStyle, CartCount, LogoImg, CartImg } from "./Header.style";
-import { useContext } from "react";
-import { CartItemsContext } from "../../context/CartItemsContext";
+import useCartItems from "@hooks/useCartItems";
+import * as H from "./Header.style";
 
-export default function Header() {
-  const { cartItems } = useContext(CartItemsContext);
-  const quantityInCart = cartItems.length;
+export default function Header({
+  handleOpenCartModal,
+}: {
+  handleOpenCartModal: () => void;
+}) {
+  const { cartItems } = useCartItems();
+  const quantityInCart = cartItems && cartItems.length;
```

Comment:
`cartItems`의 타입이 `CartItem[]`으로 추론 되는 걸 보면 아무런 아이템이 없어도 빈 배열로 항상 `true`로 평가 될텐데 어떤 의도로 이 코드를 작성하신건지 잘 이해가 되지 않네용

Reply:
cartItems이 undefined일 수 있다고 하면 `quantityInCart`의 type이 `false | number` 로 추론 될텐데 이거 역시 어색하게 느껴집니다!

Reply:
의도를 명확히 전달하려면 아래처럼 변경해봐도 좋을거 같네용
```suggestion
  const quantityInCart = cartItems.length ? cartItems.length : 0;
```